src/main/java/com/yahoo/bullet/kafka/CertRefreshingSslEngineFactory.java

```diff
@@ -0,0 +1,205 @@
+
+@Slf4j
+public class CertRefreshingSslEngineFactory implements SslEngineFactory {
+    public static final String SSL_CERT_LOCATION_CONFIG = "ssl.cert.refreshing.cert.location";
+    public static final String SSL_KEY_LOCATION_CONFIG = "ssl.cert.refreshing.key.location";
+    public static final String SSL_KEY_REFRESH_INTERVAL_CONFIG = "ssl.cert.refreshing.refresh.interval.ms";
+
+    // Package level for testing
+    String publicCertLocation;
+    String privateKeyLocation;
+    String truststoreLocation;
+    Password truststorePassword;
+    int keyRefreshInterval;
+    String[] cipherSuites = null;
+    String[] enabledProtocols = null;
+    SSLContext sslContext;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void configure(Map<String, ?> configs) {
+        log.info("Configuring {}...", this.getClass().getCanonicalName());
+        this.publicCertLocation = findFileOrThrow(configs, SSL_CERT_LOCATION_CONFIG);
+        this.privateKeyLocation = findFileOrThrow(configs, SSL_KEY_LOCATION_CONFIG);
+        this.truststoreLocation = findFileOrThrow(configs, SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+        this.truststorePassword = (Password) configs.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
+        this.keyRefreshInterval = Integer.parseInt((String) configs.get(SSL_KEY_REFRESH_INTERVAL_CONFIG));
+
+        SecurityUtils.addConfiguredSecurityProviders(configs);
+
+        List<String> cipherSuitesList = (List<String>) configs.get(SslConfigs.SSL_CIPHER_SUITES_CONFIG);
+        if (cipherSuitesList != null && !cipherSuitesList.isEmpty()) {
+            this.cipherSuites = cipherSuitesList.toArray(new String[cipherSuitesList.size()]);
+        }
+
+        List<String> enabledProtocolsList = (List<String>) configs.get(SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG);
+        if (enabledProtocolsList != null && !enabledProtocolsList.isEmpty()) {
+            this.enabledProtocols = enabledProtocolsList.toArray(new String[enabledProtocolsList.size()]);
+        }
+
+        this.sslContext = createSSLContext();
+    }
+
+    @Override
+    public SSLEngine createClientSslEngine(String peerHost, int peerPort, String endpointIdentification) {
+        SSLEngine sslEngine = createSSLEngine(peerHost, peerPort);
+
+        if (cipherSuites != null) {
+            sslEngine.setEnabledCipherSuites(cipherSuites);
+        }
+        if (enabledProtocols != null) {
+            sslEngine.setEnabledProtocols(enabledProtocols);
+        }
+
+        sslEngine.setUseClientMode(true);
+
+        SSLParameters sslParams = sslEngine.getSSLParameters();
+        sslParams.setEndpointIdentificationAlgorithm(endpointIdentification);
+        sslEngine.setSSLParameters(sslParams);
+
+        return sslEngine;
+    }
+
+    @Override
+    public SSLEngine createServerSslEngine(String peerHost, int peerPort) {
+        throw new RuntimeException(this.getClass().getCanonicalName() + " only supports client-side SSLEngines, it " +
```

Comment:
`throw new UnsupportedOperationException("This only supports client-side SSLEngines")` The stack trace should include all the class and method names and the exception signifies why clearly